File: 2022.01.16/python/5622.py
dial = {};
for i in range(15):
    dial[ord("A")+i] = 3+(i//3)
for i in range(15,19):
    dial[ord("A")+i] = 8
for i in range(19,22):
    dial[ord("A")+i] = 9
for i in range(22,26):
    dial[ord("A")+i] = 10
s = input()
result = 0
for char in s:
    result += dial[ord(char)]
print(result)

File: Boj.단계별/7_문자열/javascript/5622.js
let dial = new Map();
for (let i = 0 ; i<15 ; i++){
  dial.set(String.fromCharCode("A".charCodeAt(0)+i),parseInt(i/3)+3);
}
for (let i = 15 ; i <19 ; i++){
  dial.set(String.fromCharCode("A".charCodeAt(0)+i),8);
}
for (let i = 19 ; i<22 ; i++){
  dial.set(String.fromCharCode("A".charCodeAt(0)+i),9);
}
for (let i = 22 ; i<26 ; i++){
  dial.set(String.fromCharCode("A".charCodeAt(0)+i),10);
}

const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().split('\n')[0];
let result = 0;
for (let i = 0 ; i<input.length ; i++){
  result += dial.get(input[i]);
}
console.log(result);
